src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/FindPal.cs

```diff
@@ -0,0 +1,405 @@
+                });
+        }
+
+        public unsafe void FindByIssuerName(string issuerName)
+        {
+            fixed (char* pIssuerName = issuerName)
+            {
+                FindCore(CertFindType.CERT_FIND_ISSUER_STR, pIssuerName);
+            }
+        }
+
+        public void FindByIssuerDistinguishedName(string issuerDistinguishedName)
+        {
+            FindCore(
+                delegate(SafeCertContextHandle pCertContext)
+                {
+                    string actual = GetCertNameInfo(pCertContext, CertNameType.CERT_NAME_RDN_TYPE, CertNameFlags.CERT_NAME_ISSUER_FLAG);
+                    return issuerDistinguishedName.Equals(actual, StringComparison.OrdinalIgnoreCase);
+                });
+        }
+
+        public unsafe void FindBySerialNumber(BigInteger hexValue, BigInteger decimalValue)
+        {
+            FindCore(
+                delegate(SafeCertContextHandle pCertContext)
+                {
+                    byte[] actual = pCertContext.CertContext->pCertInfo->SerialNumber.ToByteArray();
+
+                    // Convert to BigInteger as the comparison must not fail due to spurious leading zeros
+                    BigInteger actualAsBigInteger = PositiveBigIntegerFromByteArray(actual);
+
+                    GC.KeepAlive(pCertContext);
+                    return hexValue.Equals(actualAsBigInteger) || decimalValue.Equals(actualAsBigInteger);
+                });
+        }
+
+        public void FindByTimeValid(DateTime dateTime)
+        {
+            FindByTime(dateTime, 0);
+        }
+
+        public void FindByTimeNotYetValid(DateTime dateTime)
+        {
+            FindByTime(dateTime, -1);
+        }
+
+        public void FindByTimeExpired(DateTime dateTime)
+        {
+            FindByTime(dateTime, 1);
+        }
+
+        private unsafe void FindByTime(DateTime dateTime, int compareResult)
+        {
+            FILETIME fileTime = FILETIME.FromDateTime(dateTime);
+
+            FindCore(
+                delegate(SafeCertContextHandle pCertContext)
+                {
+                    int comparison = Interop.crypt32.CertVerifyTimeValidity(ref fileTime,
+                        pCertContext.CertContext->pCertInfo);
+                    GC.KeepAlive(pCertContext);
+                    return comparison == compareResult;
+                });
+        }
+
+        public unsafe void FindByTemplateName(string templateName)
+        {
+            FindCore(
+                delegate(SafeCertContextHandle pCertContext)
+                {
+                    // The template name can have 2 different formats: V1 format (<= Win2K) is just a string
+                    // V2 format (XP only) can be a friendly name or an OID.
+                    // An example of Template Name can be "ClientAuth".
+
+                    bool foundMatch = false;
+                    CERT_INFO* pCertInfo = pCertContext.CertContext->pCertInfo;
+                    {
```

Comment:
> Why the extra scope?

Looks like it's just showing the V1 case as a self-contained block.  I just moved it :).